sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1085 @@
+     */
+    IEEE802154_CAP_IRQ_CCA_DONE,
+    /**
+     * @brief the device supports going to RX after TX done.
+     */
+    IEEE802154_CAP_TXRX,
+    /**
+     * @brief the device provides the number of retransmissions
+     */
+    IEEE802154_CAP_FRAME_RETRIES_INFO,
+} ieee802154_rf_caps_t;
+
+/**
+ * @brief IEEE802.15.4 transceiver states (not to confuse with device states)
+ */
+typedef enum {
+    IEEE802154_TRX_STATE_TRX_OFF,   /**< the transceiver state if off */
+    IEEE802154_TRX_STATE_RX_ON,     /**< the transceiver is ready to receive packets */
+    IEEE802154_TRX_STATE_TX_ON,     /**< the transceiver is ready to transmit a frame */
+} ieee802154_trx_state_t;
+
+/**
+ * @brief IEEE802.15.4 Radio HAL events
+ *
+ * To follow the IEEE802.15.4 convention, an event that responds to a Request
+ * is a confirmation (Confirm). Otherwise an Indication.
+ */
+typedef enum {
+    /**
+     * @brief the transceiver detected a valid SFD
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_RX_START cap.
+     */
+    IEEE802154_RADIO_INDICATION_RX_START,
+
+    /**
+     * @brief the transceiver received a frame and lies in the
+     *        internal framebuffer.
+     *
+     * The transceiver is in @ref IEEE802154_TRX_STATE_RX_ON state when
+     * this funcion is called, but with framebuffer protection (either
+     * dynamic framebuffer protection or disabled RX). Thus, the frame
+     * won't be overwritten before calling the @ref ieee802154_radio_read
+     * function. However, @ref ieee802154_radio_read MUST be called in
+     * order to receive new packets. If there's no interest in the
+     * frame, the function can be called with a NULL buffer to drop
+     * the frame.
+     */
+    IEEE802154_RADIO_INDICATION_RX_DONE,
+
+    /**
+     * @brief the transceiver finished sending a frame or the
+     *        retransmission procedure
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_TX_DONE cap.
+     * The upper layer should immediately call @ref
+     * ieee802154_radio_ops::confirm_transmit when on this event.
+     */
+    IEEE802154_RADIO_CONFIRM_TX_DONE,
+    /**
+     * @brief the transceiver reports that the ACK timeout expired
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_ACK_TIMEOUT.
+     */
+    IEEE802154_RADIO_CONFIRM_ACK_TIMEOUT,
+    /**
+     * @brief the CCA procedure finished
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_CCA_DONE.
+     */
+    IEEE802154_RADIO_CONFIRM_CCA,
+} ieee802154_trx_ev_t;
+
+/**
+ * @brief CSMA-CA exponential backoff parameters.
+ */
+typedef struct {
+    uint8_t min;    /**< minimum value of the exponential backoff */
+    uint8_t max;    /**< maximum value of the exponential backoff */
+} ieee802154_csma_be_t;
+
+/**
+ * @brief RX information associated to a frame
+ */
+typedef struct {
+    int16_t rssi;   /**< RSSI of the received frame */
```

Comment:
I've seen most radios use `int8_t` for RSSI - is the standard mandating 16 bit here?

Reply:
![RSSI](https://user-images.githubusercontent.com/1301112/89220480-f09e2a80-d5d1-11ea-88c9-fa181633aa0d.png)

So it's 8 bit, but with an offset.

Reply:
You are right. I will move it to 8 bit

Reply:
Would be `uint8_t` then with a -174 dBm offset - I have the feeling we printed it wrong the whole time (just taking the raw value and interpreting it as `int8_t`)

Reply:
> Would be uint8_t then with a -174 dBm offset - I have the feeling we printed it wrong the whole time (just taking the raw value and interpreting it as int8_t)

Yes, it seems we have been doing it wrong :/

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/de07050b02f45c025e815b87e7554b9832e2ea20

Reply:
@benpicco I'm trying to find the reference to your screenshot, do you recall where you took it from?

Reply:
@fjmolinas that's from [802.15.4-2020](https://sci-hub.se/10.1109/IEEESTD.2020.9144691), page 155